cli/operator/node.go

```diff
@@ -721,68 +701,57 @@ func setupEventHandling(
 		fromBlock = new(big.Int).SetUint64(fromBlock.Uint64() + 1)
 	}
 
-	// load & parse local events yaml if exists, otherwise sync from contract
-	if len(cfg.LocalEventsPath) != 0 {
-		localEvents, err := localevents.Load(cfg.LocalEventsPath)
-		if err != nil {
-			logger.Fatal("failed to load local events", zap.Error(err))
-		}
+	// Sync historical registry events from Ethereum smart contract.
+	logger.Debug("syncing historical registry events", zap.Uint64("fromBlock", fromBlock.Uint64()))
```

Comment:
This code is unreachable because `setupEventHandling` is only ever called when `len(cfg.LocalEventsPath) == 0`. Which kind of suggests we are never calling `EventHandler.HandleLocalEvents` ... which is kind of weird ?

I also don't see `cfg.LocalEventsPath` being used anywhere to actually read the contents of the file, not sure if such usage was removed in the past or just hasn't been added yet, @nkryuchkov   could you elaborate ? I'd rather clean it up while we are at it (by maybe removing `cfg.LocalEventsPath` altogether if we don't need it anymore).

Reply:
We always need to call `setupEventHandling` because it handles both on-chain and local events. `cfg.LocalEventsPath` is used by `localevents.Load(cfg.LocalEventsPath)`

Reply:
> cfg.LocalEventsPath is used by localevents.Load(cfg.LocalEventsPath)

But like I mentioned above, we never enter the branch that executes `localevents.Load(cfg.LocalEventsPath)`.

Reply:
@iurii-ssv in your PR we never do because you call `setupEventHandling` only `if !usingLocalEvents`, but on stage we always call `setupEventHandling`, which calls `localevents.Load(cfg.LocalEventsPath)` `if len(cfg.LocalEventsPath) != 0`

Reply:
Oh, you are right ... I see what's going on. On **stage** the following happens:
```go
		eventSyncer := setupEventHandling(
			cmd.Context(),
			logger,
			executionClient,
			validatorCtrl,
			metricsReporter,
			networkConfig,
			nodeStorage,
			operatorDataStore,
			operatorPrivKey,
			keyManager,
		)
		if len(cfg.LocalEventsPath) == 0 {
			nodeProber.AddNode("event syncer", eventSyncer)
		}
```
And `setupEventHandling` returning `eventSyncer` baited me into thinking "there is no need to even call `setupEventHandling` if we aren't gonna do `nodeProber.AddNode`" - and I merged these 2 under `if len(cfg.LocalEventsPath) == 0`.

Let me revert that part, and see how I can separate those things from each other (to make it clear & explicit).

Reply:
Okay, reworked this a bit, relevant commit is here https://github.com/ssvlabs/ssv/pull/1819/commits/9900a0bc9a925ec2abfce7cd5ff376f369fd61e4 and PR is ready (but I'll redeploy to stage to test it again 1 more time).

What's not 100% clear to me though is "what those local events are", I thought of it as some kind of cache - but that's unlikely to be correct because we don't "subscribe" for events that come after the last even in `cfg.LocalEventsPath` file, could you clarify (and I'll add a comment about it somewhere) ?